pkg/local/diff_test.go

```diff
@@ -254,6 +315,30 @@ func setupFiles(t testing.TB, fileList []string) string {
 	return dir
 }
 
+func getSortedFilesAndDirs(fileList []string) []string {
+	res := make([]string, 0)
+	visited := make(map[string]bool)
```

Comment:
Is there a more elegant way for de-duping an array of strings?

Reply:
You might extract PathSet to a library and change its name to StringSet.  The Go idiom for a set is a map with empty values.

For the interface of this function you could still do better, but not by much.  If you wanted to, you could do this while generating the parameter fileList instead: now you could probably know when you push into a directory - and that operation happens just once.  Not sure that this solution is worth the effort.

Reply:
Thanks for the suggestions!

Regarding a new PathSet - I'm very much in doubt if it'll be used again in the future, so having said that, I think it's more readable to leave it as it.